crm_timesheet/models/crm_phonecall.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# See README.rst file on addon root folder for license details
+
+from openerp import models, fields, api, _
+from openerp.fields import DATE_LENGTH
+from datetime import datetime
+from openerp.exceptions import ValidationError
+
+
+class CrmPhonecall(models.Model):
+    _inherit = 'crm.phonecall'
+
+    analytic_account_id = fields.Many2one(
+        comodel_name='account.analytic.account')
+    timesheet_ids = fields.One2many(comodel_name='hr.analytic.timesheet',
+                                    inverse_name='phonecall_id')
+
+    def _timesheet_prepare(self, vals):
+        """
+            param: vals
+            type: dict
+            desc: crm.phonecall vals dict (It's like create and write methods)
+        """
+        if len(self) > 0:
+            self.ensure_one()
+        date = vals.get('date', self.date)
+        user_id = vals.get('user_id', getattr(self.user_id, 'id', False))
```

Comment:
I still insists that this can be simplified to `user_id = vals.get('user_id', self.user_id.id)`